retrieval/discovery/consul.go

```diff
@@ -272,9 +273,11 @@ func (cd *ConsulDiscovery) watchService(srv *consulService, ch chan<- *config.Ta
 				ConsulTagsLabel:          clientmodel.LabelValue(tags),
 			})
 		}
+
 		cd.mu.Lock()
 		select {
 		case <-srv.done:
+			cd.mu.Unlock()
```

Comment:
How about `defer cd.mu.Unlock()` right after the lock, and remove the second unlock instance below?